MovingBlocks/Terasology
# Render Task List Generator refactor / Improved Logger Output

@tdgunes: you might want to have a look at this one.
@indianajohn: it might be interesting for you too, to increase your knowledge of the system.

### Contains
* Considerably refactored the RenderTaskListGenerator class, responsible for taking the nodes in the render graph (currently just a list) and generating the list of tasks that are executed every frame to generate the image shown on screen. Also added some javadocs.
* Changed toString() methods in a number of classes to generate a better logger output, with aligned text fields separated by colons.
* Added some statistics printout at the end of the tasklist generation process.

### How to test
* Load up a game and then check the logger output for two lists: the Intermediate Renderer List first and the Renderer Task List next. The first lists all potential opengl state changes caused by each node. The second lists tasks reflecting the actual state changes taking place every frame, some state changes having been eliminated because redundant with previous or following state changes. 
* Changing the video settings in some cases remove nodes from the lists - a feature, not a bug. However there is no simple relationship between video settings and nodes as some settings enable shader features that have no corresponding node. So, you might sometimes turn something off and the lists do not change.

### Known issues
This PR doesn't solve a number of existing issues in the rendering engine but improves one of the main diagnostic tool, the logger output describing the current task list. Now I expect to be able to begin addressing issues.
